src/server/clientiface.cpp

```diff
@@ -435,32 +418,33 @@ void RemoteClient::GotBlock(v3s16 p)
 
 void RemoteClient::SentBlock(v3s16 p)
 {
-	if (m_blocks_sending.find(p) == m_blocks_sending.end())
-		m_blocks_sending[p] = 0.0f;
-	else
+	if (!m_blocks_sending.insert(p).second)
 		infostream<<"RemoteClient::SentBlock(): Sent block"
 				" already in m_blocks_sending"<<std::endl;
 }
 
 void RemoteClient::SetBlockNotSent(v3s16 p)
 {
-	m_nothing_to_send_pause_timer = 0;
-
 	// remove the block from sending and sent sets,
-	// and mark as modified if found
-	if (m_blocks_sending.erase(p) + m_blocks_sent.erase(p) > 0)
-		m_blocks_modified.insert(p);
+	// and reset the scan loop if found
+	if (m_blocks_sending.erase(p) + m_blocks_sent.erase(p) > 0) {
+		m_nothing_to_send_pause_timer = 0;
+		if (!m_perf_tradeoffs) {
```

Comment:
Server will get to the block in the next full round if performance tradeoffs are enabled.